### PR TITLE
feat(masthead-v2.1): simple megamenu (fka "listing") design updates

### PR DESCRIPTION
**Pull request by jkaeser**
### Description

Updates component code and styles to make "simple" megamenus comply with the 2.1 design specifications.

See the design & functional specs: https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=3205%3A192891

### To Review

Visit the deploy preview for the Masthead using v2 data (https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/9627/index.html?path=/story/components-masthead--with-v-2-data) and ensure the updates match the design & functional specs linked above and don't break the usability of the navigation.

**Note:** updates to the View All links will be handled in a separate ticket.

### Changelog

**New**

- Adds support for headline area to simple megamenus. This includes an optionally linked heading and description text.
- Reintroduces support for highlighted column to simple megamenus.

**Changed**

- Consolidates redundant renderer logic for category headings.
- Reduce logical complexity for applying appropriate design treatments to `DDSMegamenuCategoryLink` components. Instead of requiring adopters to declare a design treatment, the component styles react to content entering the component. This makes the component more predictable and flexible.
- Update design tokens for typography and spacing throughout relevant components.

### CODE REVIEW

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 1]
```diff
@@ -140,10 +141,10 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
    * @param sections menu section data object
    */
   // eslint-disable-next-line class-methods-use-this
-  protected _getHighlightedMenuItems(sections) {
+  protected _getMenuItems(sections) {
```
Comment (andy-blum) [inline packages/web-components/src/components/masthead/masthead-composite.ts:144]: I think renaming this function may have broken the "selected menu item" highlighting.
Screenshot 2022-12-05 at 1 52 05 PM

File: packages/web-components/src/components/masthead/megamenu-heading.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  protected _handleSlotChange(event) {
+    this._hasContent = (event.target as HTMLSlotElement).assignedNodes().filter(child => child?.textContent?.trim()).length > 0;
+  }
+
+  render() {
+    const hasHref = Boolean(this.href);
+    return html`
+      <h2 ?data-has-href="${hasHref}">
```
Comment (andy-blum) [inline packages/web-components/src/components/masthead/megamenu-heading.ts:72]: Should we add a way to control the heading level dynamically? Right now everything loads in as an H2 which implies they're all equally important in the hierarchy, but the visual styling and layout don't suggest that they should be.
Screenshot 2022-12-05 at 1 49 04 PM

**Review by andy-blum — CHANGES_REQUESTED** [review]
This looks pretty good - a couple of things I'm noticing, though:
- The content when fully scrolled needs to clear our overflow indicator shadow (see below)
- We might have some a11y issues with all the H2's
- The selected item highlight isn't working

Screenshot 2022-12-05 at 1 38 19 PM

File: packages/web-components/src/components/masthead/megamenu-heading.ts (hunk 2 above)
Comment (jkaeser) [reply packages/web-components/src/components/masthead/megamenu-heading.ts]: Good catch. I added a pretty rudimentary way to control this in [ecf24a3](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9627/commits/ecf24a38f7c4a60d1c5bb36adc320396356ed3b5).

**Review by andy-blum — APPROVED** [review]
LGTM

**Review by QALosey — APPROVED** [review]
This looks good to me.

**Review by KiBeta — APPROVED** [review]
Looks good to me as well.

**Comment by jwitkowski79** [comment]
@jkaeser A few design issues I'm finding with spacing. You can find the specs here [here:](https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=3179%3A173923&t=BVnd1RiiPisGO9wY-1)

**Spacing**

- Spacing between the Heading level 1 with description text and the link list below needs to be updated to 24px.
- Spacing between the H2 and link list needs to be updated to 24px (I think this is already 24px but wasn't able to get the value. Just double check this please and thank you! )

BB7FB753-AF9B-435D-B4E1-25BC372F1D3D

- Spacing between the link list containers needs to be updated to 24px
CCD83B7C-73B6-4EA8-8D30-051B7CB91BCD

**Divider lines**

- Divider lines should extend full width
643829FE-1D5E-43FF-AC1C-5B6A72C04184

EB54BAE5-C58C-45BB-A163-CF9EF7257A16

**Comment by jkaeser** [comment]
@jwitkowski79 Thanks for the review! I've addressed the spacing problems, and the divider lines will be handled in a separate issue for the View All link changes.

Ready for you to take another look when you have the chance.

**Review by jwitkowski79 — APPROVED** [review]
This looks good. Thank you @jkaeser

**Pull request merged**